clients/bolt-sdk/src/world/transactions.ts

```diff
@@ -62,28 +60,31 @@ export async function InitializeNewWorld({
 export async function AddEntity({
   payer,
   world,
+  seed,
   connection,
 }: {
   payer: PublicKey;
   world: PublicKey;
+  seed?: string;
   connection: Connection;
-}): Promise<{ transaction: Transaction; entityPda: PublicKey; entityId: BN }> {
+}): Promise<{ transaction: Transaction; entityPda: PublicKey }> {
   const worldInstance = await World.fromAccountAddress(connection, world);
-  const entityId = new BN(worldInstance.entities);
-  const entityPda = FindEntityPda(new BN(worldInstance.id), entityId);
-
+  const entityPda = FindEntityPda(
+    new BN(worldInstance.id),
+    new BN(worldInstance.entities),
+    seed
+  );
   const createEntityIx = createAddEntityInstruction(
     {
       world,
       payer,
       entity: entityPda,
     },
-    { extraSeed: null }
+    { extraSeed: seed ?? null }
   );
   return {
     transaction: new Transaction().add(createEntityIx),
     entityPda,
-    entityId,
```

Comment:
removing this id because this might not be correct if the user passed a seed as parameter.